packages/datatrak-web/src/api/mutations/useEditUser.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { useMutation, useQueryClient } from 'react-query';
+import { Project } from '@tupaia/types';
+import { put } from '../api';
+import { useUser } from '../queries';
+
+type ProjectId = { projectId: Project['id'] };
+
+export const useEditUser = onSuccess => {
+  const queryClient = useQueryClient();
+  const { data: user } = useUser();
+
+  return useMutation<any, Error, ProjectId, unknown>(
+    async ({ projectId }: ProjectId) => {
+      if (!user?.id || !projectId) {
+        return;
```

Comment:
Should we put an error message or something in here so it fails more 'loudly'?

Reply:
Good point. I've made a new ticket for setting up this kind of messaging ui [here](https://linear.app/bes/issue/WAITP-1437/toast-messages)